With UseMaui = true in class library I get a '"CopyLocalFilesOutputGroup" does not exist' error

### Description

When I create a class library for reusable maui components and add ```<UseMaui>true</UseMaui>``` to the csproj I get this error: 
```C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets(188,5): error MSB4057: The target "CopyLocalFilesOutputGroup" does not exist in the project.  C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets``` 

Used rc2 and VS preview 5

I am not sure if I am supposed to use ```<UseMaui>true</UseMaui>``` for this purpose. I would like to know the alternative way if there is one. Still there may have to be a better error in this situation.

Screenshot of the situation:
![image](https://user-images.githubusercontent.com/963462/165458258-91fa12e2-f771-4744-ba89-b40bf72e3585.png)

Minimal reproduction:
[CopyLocalFilesOutputGroup-does-not-exist.zip](https://github.com/dotnet/maui/files/8569575/CopyLocalFilesOutputGroup-does-not-exist.zip)



### Steps to Reproduce

1. Create a new maui app
2. Create a new class library
3. Add <UseMaui>true</UseMaui> to the csproj of the class library
4. Add a reference to the class library in the maui app
5. Compile.



### Version with bug

Release Candidate 2

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

It is a compile problem so I guess it affects all platforms with RC 2. I compiled on Windows.

### Did you find any workaround?

No

### Relevant log output

```shell
Build started...
1>------ Build started: Project: ClassLibrary1, Configuration: Debug Any CPU ------
1>You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
1>Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
1>ClassLibrary1 -> C:\Users\pauld\source\repos\MauiApp3\ClassLibrary1\bin\Debug\net6.0\ClassLibrary1.dll
2>------ Build started: Project: MauiApp3, Configuration: Debug Any CPU ------
2>Skipping analyzers to speed up the build. You can execute 'Build' or 'Rebuild' command to run analyzers.
2>MauiApp3 -> C:\Users\pauld\source\repos\MauiApp3\MauiApp3\bin\Debug\net6.0-windows10.0.19041.0\win10-x64\MauiApp3.dll
2>C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets(188,5): error MSB4057: The target "CopyLocalFilesOutputGroup" does not exist in the project.  C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets
2>Done building project "ClassLibrary1.csproj" -- FAILED.
========== Build: 1 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Deploy: 0 succeeded, 0 failed, 0 skipped ==========
```

This does not work if you have a class library that uses only 
`<UseMaui>true</UseMaui>`
(no platform specific code) and is referenced by a Maui application. The building of the library then fails with

`C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets(188,5): error MSB4057: The target "CopyLocalFilesOutputGroup" does not exist in the project.
`

Repro steps:

- Create a Maui Application. 
- Add a Maui class library
- Reference the class library from the application
- Modify the csproj of the class library as follows:

```
<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFrameworks>net6.0</TargetFrameworks>
		<UseMaui>true</UseMaui>
		<ImplicitUsings>enable</ImplicitUsings>
	</PropertyGroup>

</Project>

```
Build the project. The build of the library will fail with the above error but **only** if it is referenced from the main project. If it is not referenced the build succeeds.

I created all of my MAUI libs from scratch with RC2 and copied the class files over. The CSPROJ is as per [shugaoye](https://github.com/shugaoye) post. However. I cannot build the main project (i.e. that references the 5 MAUI libraries).

C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets(188,5): error MSB4057: The target "CopyLocalFilesOutputGroup" does not exist in the project.  C:\Program Files\dotnet\packs\Microsoft.Maui.Sdk\6.0.300-rc.2.5513\Sdk\WinUI.NetCore.targets

This happens for two of my 5 MAUI Libs. If I compile them only, then the compile succeeds.

Currently dead in the water. :(

--- a/.nuspec/Microsoft.Maui.Controls.SingleProject.props
+++ b/.nuspec/Microsoft.Maui.Controls.SingleProject.props
@@ -2,11 +2,7 @@
 
   <PropertyGroup Condition=" '$(SingleProject)' == 'true' and '$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'windows' ">
     <OutputType Condition="'$(OutputType)' == 'Exe'">WinExe</OutputType>
-    <EnableMsixTooling Condition="'$(EnableMsixTooling)' == ''">true</EnableMsixTooling>
-    <EnablePreviewMsixTooling Condition="'$(EnablePreviewMsixTooling)' == ''">$(EnableMsixTooling)</EnablePreviewMsixTooling>
     <WindowsPackageType Condition=" '$(WindowsPackageType)' == '' and '$(EnableMsixTooling)' == 'true' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">MSIX</WindowsPackageType>
-    <WinUISDKReferences Condition=" '$(WinUISDKReferences)' == '' and '$(EnableMsixTooling)' == 'true' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe') ">false</WinUISDKReferences>
-    <GenerateLibraryLayout Condition=" '$(GenerateLibraryLayout)' == '' and '$(EnableMsixTooling)' == 'true' and '$(OutputType)' != 'Exe' and '$(OutputType)' != 'WinExe' ">true</GenerateLibraryLayout>
     <WindowsAppSdkBootstrapInitialize Condition=" '$(WindowsAppSdkBootstrapInitialize)' == '' and '$(EnableMsixTooling)' == 'true' and '$(OutputType)' != 'Exe' and '$(OutputType)' != 'WinExe' ">false</WindowsAppSdkBootstrapInitialize>
     <PublishAppXPackage Condition=" '$(PublishAppXPackage)' == '' and '$(EnableMsixTooling)' == 'true' and '$(WindowsPackageType)' == 'MSIX' ">true</PublishAppXPackage>
     <_SingleProjectRIDRequired Condition="'$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe'">true</_SingleProjectRIDRequired>

--- a/src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.After.targets
+++ b/src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.After.targets
@@ -5,6 +5,12 @@
     <MSBuildWarningsAsMessages Condition=" '$(TargetPlatformIdentifier)' == 'Android' ">$(MSBuildWarningsAsMessages);XA4218</MSBuildWarningsAsMessages>
   </PropertyGroup>
 
-  <Import Project="WinUI.targets" Condition=" '$(TargetPlatformIdentifier)' == 'windows' " />
+  <!--
+    This double check seems excessive, but importing the WinUI.targets when building for windows is not
+    sufficient. The WASDK targets assume everything is WinUI and thus just passes along the TFM of the app.
+    As a result, if you have a net6.0 class library, the app will call MSBuild on that library - with the Windows TFM!
+    This results in the $(TargetPlatformIdentifier) condition being met - even though there are no WASK targets to run!
+  -->
+  <Import Project="WinUI.targets" Condition=" '$(TargetPlatformIdentifier)' == 'windows' and '$(WindowsAppSDKWinUI)' == 'true'" />
 
 </Project>

--- a/src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.targets
+++ b/src/Workload/Microsoft.Maui.Sdk/Sdk/Microsoft.Maui.Core.Sdk.targets
@@ -4,6 +4,14 @@
     <UsingMicrosoftMauiCoreSdk>true</UsingMicrosoftMauiCoreSdk>
   </PropertyGroup>
 
+  <!-- Set some property defaults that MAUI requires - especially for dotnet build to work -->
+  <PropertyGroup Condition=" '$([MSBuild]::GetTargetPlatformIdentifier($(TargetFramework)))' == 'windows' ">
+    <EnableMsixTooling Condition=" '$(EnableMsixTooling)' == '' ">true</EnableMsixTooling>
+    <EnablePreviewMsixTooling Condition=" '$(EnablePreviewMsixTooling)' == '' ">$(EnableMsixTooling)</EnablePreviewMsixTooling>
+    <GenerateLibraryLayout Condition=" '$(GenerateLibraryLayout)' == '' and '$(EnableMsixTooling)' == 'true' and '$(OutputType)' != 'Exe' and '$(OutputType)' != 'WinExe' ">true</GenerateLibraryLayout>
+    <WinUISDKReferences Condition=" '$(WinUISDKReferences)' == '' and '$(EnableMsixTooling)' == 'true' ">false</WinUISDKReferences>
+  </PropertyGroup>
+
   <!-- Imported last -->
   <PropertyGroup>
     <AfterMicrosoftNETSdkTargets>$(AfterMicrosoftNETSdkTargets);$(MSBuildThisFileDirectory)Microsoft.Maui.Core.Sdk.After.targets</AfterMicrosoftNETSdkTargets>

--- a/src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.NetCore.targets
+++ b/src/Workload/Microsoft.Maui.Sdk/Sdk/WinUI.NetCore.targets
@@ -355,7 +355,7 @@
 
   </Target>
 
-<Target Name="GetPriOutputs"
+  <Target Name="GetPriOutputs"
           Condition="'$(AppxGeneratePriEnabled)'=='true'"
           Returns="@(PriOutputs);@(ProjectPriFile)"
           DependsOnTargets="$(GetPriOutputsDependsOn)">

